extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -750,6 +781,45 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
+   * Refreshes ad slot by fetching a new creative and rendering it. This leaves
+   * the current creative displayed until the next one is ready.
+   *
+   * @param {function()} refreshEndCallback When called, this function will
+   *   restart the refresh cycle.
+   * @return {Promise} A promise that resolves when all asynchronous portions of
+   *   the refresh function complete. This is particularly handy for testing.
+   */
+  refresh(refreshEndCallback) {
+    return new Promise(resolve => {
```

Comment:
I'd have to wrap the timerFor block in a new Promise at the least. I can do that if you find it preferable.